### PR TITLE
[WIP] Bugsnax Genetics: Splice yourself with dead mobs for powers

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

Adds the Splicing Machinery to the Genetics Lab.
Put in a supported dead simple animal/basic mob corpse and a humanoid to splice onto, and it'll splice the mob on as a limb with a power!

Moves Cytology to Genetics.
You can now stick a simple animal/basic mob corpse into the reagent grinder for plumbing to get 100 protein.
Mining, icebox, and any other Splicing supported creature will be reproducible with Cytology for protein.

In the event of somehow losing all your cytology swabs, your three starter corpses, and all the miners who can bring back corpses, a lootbox can be purchased from Cargo with a random biome compatible corpse in it.

## Why It's Good For The Game

Genetics as it stands right now hasn't innovated in a decade or two and really needs new features.
So, bugsnax genetics.
Splice in mob limbs. Get cool powers.
The more limbs of a specific type, the stronger that power is.
## Todo list:

- [X] Stickmen: Makes you face south at all times no matter what.
- [ ] Goliaths: Lets you grab people and items to pull them towards you from a distance with a tentacle. If it hits a wall or empty space, pulls you towards that wall or empty space. More limbs mean longer reach.
- [ ] Legions: Sacrifice a limb(or multiple limbs) to explode into legion skulls that attack anyone you attack or anyone who attacks you while following you around.
- [ ] Brimdemons: Burn nutriment to vomit a deadly laser in a straight line that's stronger and penetrates more surfaces the more limbs you have.
- [ ] Watchers: Freeze a cone in front of you, making the floor slippery and dangerous. Wider cone the more limbs you have.
- [ ] Goldgrubs: Provides a permanent built in source of storage. The more limbs, the bigger the items that can fit in that storage. Storage drops everything to the floor if you die.
- [ ] Polar Bears: Activate a rage mode that gives you undroppable bear claws, cuts your melee cooldown by half, and lets out a loud roar. The more limbs, the longer the rage mode lasts.
- [ ] Wolves: Let out a howl and immediately go to 120% speed negating any slowdown from any source for a short period of time. When you come out of it, you'll be slowed down to 80% speed, on top of any other slowdowns you might be experiencing. The more limbs, the shorter the slow period.
- [ ] Ash Drake: ???????????
- [ ] Bubblegum: ???????????
- [ ] Wendigo: ??????????????
- [ ] Colossus: ??????????????
- [ ] Bubblegum: ????????????
- [ ] Legion: ?????????????????

Maybe more down the line.

## Changelog
wip, ask me later

### CODE REVIEW

**Comment by ATH1909** [comment]
okay but how do they GET the mobs

**Comment by castawaynont** [comment]
> okay but how do they GET the mobs

drag the corpses from mining I suppose

File: code/modules/genetics/machinery/splicer.dm [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+		BODY_ZONE_HEAD,
+		BODY_ZONE_CHEST,
+		BODY_ZONE_L_ARM,
+		BODY_ZONE_R_ARM,
+		BODY_ZONE_L_LEG,
+		BODY_ZONE_R_LEG,
+	)
+	for(var/body_part_zone in body_zones_to_check)
+		var/obj/item/bodypart/body_part = splice_target.get_bodypart(body_part_zone)
+		if(body_part.limb_id in GLOB.standard_limb_types)
```
Comment (Kapu1178) [inline code/modules/genetics/machinery/splicer.dm:75]: get_bodypart() returns null when no bodypart is in that slot

File: code/modules/genetics/machinery/splicer.dm [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+			applicable_bodyparts += body_part // We want to only replace parts that haven't been genesliced yet.
+	if(!length(applicable_bodyparts))
+		applicable_bodyparts = body_zones_to_check // Gotta replace something at this point.
+	var/obj/item/bodypart/new_part = new GLOB.splice_results[corpse.type][pick(applicable_bodyparts)]
```
Comment (Kapu1178) [inline code/modules/genetics/machinery/splicer.dm:79]: byond is usually not very nice about doing `new list[index]`, set it to a var before hand

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Comment by moocowswag** [comment]
Will you be able to buy chromosome boosted injectors via your uplink?

**Comment by Serxule** [comment]
this is just powercreep except extremely obvious

**Comment by YuliceTheDruid** [comment]
You may as well just put a genetics lab on lavaland and remove it from the sci lobby if you really wanna go through with this, but to me this smacks largely of "I can't remove hulk because headcoders decreed we can't remove it entirely until we rework gene, so I'm going to slap something together as fast as I can." And again, I see less Genetics work in this and the thing tralezab is working on and more medical work than anything, it's just organ swapping/limb splicing but with a fancy machine so you can call it science.

**Comment by Gboster-0** [comment]
has potential, but it would probably need a lab down on lavaland that only geneticists can access and maybe an alternate way to get mobs besides cargo/xenobiology business

**Comment by Shadowflame909** [comment]
Powers are cool. Reminds me of the attempt to action button all abilities in game

**Comment by GuillaumePrata** [comment]
For all the "move genetics to lavaland comments", it would be easier to let it stay on station and get the ball rolling on limbs from normal mobs that cytology can spawn already, as cytology is being moved into genetics.

Cytology has a good list of mobs right now and it is stupidly easy to expend cytology code wise.
We would just be bottlenecked by how adding limbs work with this PR, which doesn't look complicated at all outside of the sprites.

**Comment by Ryusenshu** [comment]
"Removes the DNA consoles from Genetics."
And with that alone, i dont like this PR one single bit

**Comment by Blevruz** [comment]
cool concept! not a replacement for genetics.
keep genetics, label this as a cytology expansion, and you're good.

**Review by Timberpoes — CHANGES_REQUESTED** [review]
I do not think this should be merged, and here's my justification:

Science already has very little content that isn't on the chopping block.

Xenobio is also in a massive limbo area as well where it cannot be expanded and some would rather it be removed.

I don't want to see genetics as-is removed entirely. Any content I'd like to see would be an extension to genetics so it isn't so reliant on ATCG and new mutations for content.

If the new content is so superior to ATCG genetics, it can supplement or even entirely replace ATCG genetics **in the future** if deemed necessary.

Or alternatively it can pave the way for more radical or fundamental changes to ATCG genetics that aren't possible now for risk of gutting another department from Science.

The new content doesn't have to be aimed to replace genetics at all, and can simply expand and enhance the job to extend the existing gameplay loops. Ideally it's also content that's less problematic to expand (I.e. - Adding or buffing mutations right now often requires combatting the maintainer team or, in some circumstances like racial mutations, the headcoders.)

As a result I'm going to be drawing a line that whatever this PR intends to do, **at this point in time it should not replace ATCG genetics** unless and until the other maintainers band together and tell me I'm dent-headed, or a really fucking amazing design document appears that the maint team is satisfied stands the test of a brand new standalone job or role that they're willing to remove existing genetics entirely as a result.

**Comment by celulamp** [comment]
If you just kept the dna console and kept this as extra content you can do as geneticist this would prolly get accepted

which is why I think armhulens got accepted

which is kinda sad I like most of the mob abilities and cytology being moved to genetics since it would make a lot of sense

**Comment by Iamgoofball** [comment]
i have removed the part about removing genetics as-is, please remove the DNM

**Comment by Timberpoes** [comment]
Since this no longer removes current genetics ATCG content, I'm now happy to remove the DNM.

If genetics gets enough new, original, fun and balancable content to fully replace the power fantasy joy of ATCG and the genetics endgame, then that status quo can be looked at.

Alternatively, if genetics actually gets enough interesting genetic content not relating to mutations and ATCG, genetics as a whole may reach the point that we can lock the ATCG/mutation part behind some sort of macro-economy (e.g the resource economy, the money economy, or any other economy) and rebalancing (BUFF!) them by their new scarcity while genetics as a whole is still a job with the content to justify its position as a full Science department.

**Comment by optimumtact** [comment]
This doesn't even compile and looks like you threw it together in about 30 minutes after having a thought.

Maybe PR it when it's actually ready and you have more than one feature set

**Comment by dwasint** [comment]
should of been drafted over closing IMO, i think the idea is novel but needs more to it

**Comment by optimumtact** [comment]
A drafted pr should be nearly ready for merge, this isn't even close and there's nothing substantive to actually code review.

**Comment by Iamgoofball** [comment]
oranges is right, ye

**Pull request closed without merging**